edifice/_component.py

```diff
@@ -522,8 +530,13 @@ def render(self):
         """
         raise NotImplementedError
 
+P = tp.ParamSpec("P")
+
+def not_ignored(arg: tuple[str, tp.Any]) -> bool:
+    return arg[0][0] != "_"
 
-def make_component(f):
+# TODO: Should we really allow the function to return `Any`?
```

Comment:
The tests rely on being able to return an arbitrary value from the function. This is what gets returned from the `render()` method of the created component. I think it'd be nicer to assert that the function can only return `Component | None` but that might be better to do in a future PR.